code/modules/mob/living/carbon/brain/brain_item.dm

```diff
@@ -69,23 +68,36 @@
 	owner.brain_op_stage = 4.0
 
 	var/obj/item/organ/brain/B = src
-	if(istype(B) && istype(owner))
+	if(istype(B) && istype(owner) && owner.internal_organs_by_name[organ_tag] == src)

```

Comment:
Fair enough. I'll see if I can make sense of this monstrosity to where I can find something that works, then